### PR TITLE
[CALCITE-4772]PushProjector should retain alias when handling RexCall

### PR DESCRIPTION
**Pull request by YuKongEr**
For  [CALCITE-4772](https://issues.apache.org/jira/projects/CALCITE/issues/CALCITE-4772?filter=allissues)

### CODE REVIEW

**Comment by zabetak** [comment]
Hey @YuKongEr  there are some CI failures. Can you please fix those?

**Comment by YuKongEr** [comment]
@zabetak thanks, i will fix up

**Comment by rubenada** [comment]
@YuKongEr there seems to be a CheckerFramework error.
Tip: if I am not mistaken, you can run CheckerFramework locally (to verify before commit/push) via `gradlew build -x test -PenableCheckerframework`

**Comment by YuKongEr** [comment]
@rubenada oh, get it

File: core/src/main/java/org/apache/calcite/rel/rules/PushProjector.java [hunk 1]
```diff
@@ -562,18 +562,30 @@ public Project createProjectRefsAndExprs(
...
+      String alias = SqlUtil.deriveAliasFromOrdinal(preserveExpOrdinal++);
+      RelDataTypeField relDataTypeField = getRelDataTypeField(projExpr);
+      if (relDataTypeField != null) {
+        alias = relDataTypeField.getName();
+      }
       newProjects.add(
-          Pair.of(
-              newExpr,
-              SqlUtil.deriveAliasFromOrdinal(preserveExpOrdinal++)));
+          Pair.of(newExpr, alias));
```
Comment (zabetak) [inline core/src/main/java/org/apache/calcite/rel/rules/PushProjector.java:571]: How about the following refactoring?
```
      final String originalFieldName = findOriginalFieldName(projExpr);
      final String newAlias;
      if(originalFieldName != null) {
        newAlias = originalFieldName;
      } else {
        newAlias = SqlUtil.deriveAliasFromOrdinal(preserveExpOrdinal);
      }
      newProjects.add(Pair.of(newExpr, newAlias));
      preserveExpOrdinal++;
```

File: core/src/main/java/org/apache/calcite/rel/rules/PushProjector.java [hunk 2]
```diff
@@ -562,18 +562,30 @@ public Project createProjectRefsAndExprs(
...
     }
 
     return (Project) relBuilder.push(projChild)
         .projectNamed(Pair.left(newProjects), Pair.right(newProjects), true)
         .build();
   }
 
+  private @Nullable RelDataTypeField getRelDataTypeField(RexNode originRexNode) {
```
Comment (zabetak) [inline core/src/main/java/org/apache/calcite/rel/rules/PushProjector.java:579]: Rename method to `findOriginalFieldName`?

File: core/src/main/java/org/apache/calcite/rel/rules/PushProjector.java [hunk 3]
```diff
@@ -562,18 +562,30 @@ public Project createProjectRefsAndExprs(
...
+    if (origProj == null) {
+      return null;
+    }
+    int idx = origProj.getProjects().indexOf(originRexNode);
+    if (idx < 0) {
+      return null;
+    }
+    return origProj.getRowType().getFieldList().get(idx);
```
Comment (zabetak) [inline core/src/main/java/org/apache/calcite/rel/rules/PushProjector.java:587]: Return the name of the field directly?

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 4]
```diff
@@ -3481,6 +3481,21 @@ private void checkLiteral2(String expression, String expected) {
     sql(sql).optimize(rules, null).ok(expected);
   }
 
+  @Test void testMultiplicationKeepAliased() {
```
Comment (zabetak) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:3484]: Rename to: `testMultiplicationRetainsExplicitAlias`

**Review by zabetak — COMMENTED** [review]
Overall LGTM, only some minor comments.

**Comment by xy2953396112** [comment]
Could you update the title `PushProjector should retain alias when handling RexCall`?

**Comment by xy2953396112** [comment]
> Overall LGTM, only some minor comments.

+1

**Comment by xy2953396112** [comment]
Please fix CI failures.

**Comment by YuKongEr** [comment]
@xy2953396112 done

**Pull request closed without merging**